server_test.go

```diff
@@ -130,3 +130,27 @@ func (s) TestStreamContext(t *testing.T) {
 		t.Fatalf("GetStreamFromContext(%v) = %v, %t, want: %v, true", ctx, stream, ok, expectedStream)
 	}
 }
+
+func (s) TestContextPropagationForChainUnaryServerInterceptor(t *testing.T) {
+	parentKey := ctxKey("parentKey")
+
+	firstInt := func(ctx context.Context, req interface{}, info *UnaryServerInfo, handler UnaryHandler) (interface{}, error) {
+		if ctx.Value(parentKey) == nil {
+			t.Fatalf("first interceptor should have %v in context", parentKey)
+		}
+
+		return handler(ctx, req)
+	}
+
+	testHandler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return nil, nil
+	}
+
+	server := NewServer(ChainUnaryInterceptor(firstInt))
```

Comment:
I think we wouldn't need this test. This is basically testing that `firstInt` is called. This should already be covered by the more comprehensive end2end tests.